src/Aspire.Hosting.Azure.CognitiveServices/Aspire.Hosting.Azure.CognitiveServices.csproj

```diff
@@ -10,6 +10,7 @@
 
   <ItemGroup>
     <Compile Include="$(RepoRoot)src\Shared\AzureRoleAssignmentUtils.cs" />
+    <Compile Include="$(ComponentsDir)\Common\ArgumentExceptionExtensions.cs" Link="ArgumentExceptionExtensions.cs" />
```

Comment:
Instead of doing this in each project file, I suggest considering a different approach.

* If this file needs to be linked in each project, then do this injection at Directory.Build.props/targets (or alike).
* If this file needs to be conditionally linked in project, then do a conditional injection at Directory.Build.props/targets (or alike) controlled by an opt-in switch in each project.
For example: https://github.com/dotnet/extensions/blob/1e4f6c51798be661b8d8743dd464ca9f54b0c9a3/src/Libraries/Microsoft.AspNetCore.HeaderParsing/Microsoft.AspNetCore.HeaderParsing.csproj#L15C6-L15C41